tests/e2e/performance/test_performance_testing.py

```diff
@@ -1,55 +1,504 @@
-from os import path
+import multiprocessing
+import shutil
+import sys
+from time import time
 from pathlib import Path
 from string import Template
-from subprocess import run
+from contextlib import contextmanager
+from typing import List, Optional, Tuple, Generator
 
 import pytest
+from starkware.starknet.services.api.contract_class import ContractClass
 
-from tests.e2e.conftest import ACTUAL_CWD
+from protostar.commands.test.test_cases import PassedTestCase
+from protostar.commands.test.test_collector import TestCollector
+from protostar.commands.test.test_runner import TestRunner
+from protostar.commands.test.test_shared_tests_state import SharedTestsState
+from protostar.commands.test.test_suite import TestSuite
+from protostar.utils.compiler.pass_managers import ProtostarPassMangerFactory
+from protostar.utils.starknet_compilation import StarknetCompiler, CompilerConfig
 
-TEST_SUITE_HEADER = """
-%lang starknet
+SCRIPT_DIRECTORY = Path(__file__).parent
 
-from starkware.cairo.common.cairo_builtins import HashBuiltin
 
-@storage_var
-func target() -> (res : felt):
-end
+def _multiply_cases(test_body: str) -> Tuple[str, List[str]]:
+    source_code = ""
+    cases_names = []
+    for i in range(15):
+        case_name = f"test_case_{i}"
+        source_code += Template(
+            """
+            @external
+            func $case_name{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}():
+                $body
+                return ()
+            end
+            """
+        ).substitute(case_name=case_name, body=test_body)
+        cases_names.append(case_name)
+    return source_code, cases_names
 
-"""
 
-TEST_CASE_TEMPLATE = Template(
+def make_test_file(test_body: str, imports: Optional[str] = "") -> Tuple[str, List[str]]:
+    source, cases = _multiply_cases(test_body)
+    return (
+        f"""
+        %lang starknet
+        from starkware.cairo.common.cairo_builtins import HashBuiltin
+        {imports}
+        {source}
+    """,
+        cases,
+    )
+
+
+@pytest.fixture(name="tmp_test_dir")
+def make_test_dir() -> Generator[Path, None, None]:
+    test_dir_path = SCRIPT_DIRECTORY / "tmp_test_files"
```

Comment:
Don't we have already a functionality for this? 🤨

Reply:
I don't think so - if it exists you can point me to one

Reply:
https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html